itunes_app_scraper/scraper.py

```diff
@@ -153,7 +153,7 @@ def get_similar_app_ids_for_app(self, app_id, country="nl", lang="nl"):
 
 		return ids
 
-	def get_app_details(self, app_id, country="nl", lang="", flatten=True):
+	def get_app_details(self, app_id, country="nl", lang="", flatten=True, sleep=None):
```

Comment:
this method already has a sleep (when the first request fails)
since this method normally executes a single request - I don't like the idea to pass a sleep parameter into this method - the sleep should be done outside of the scope of this method e.g.

```
scraper.get_app_details
sleep
scraper.get_app_details
```

this way the control flow is much clearer

Reply:
If the request is blocked due to too many requests then it just returns an empty response - at least for the ratings. That cannot be distinguished from the regular empty response if no data is available, e.g. if there do not exist any ratings for the app in the provided country. Thus there does not happen any exception.

By offensive presentation in the get_app_details() parametrization and documentation we might sensitize the user about the sleep usage and to prevent him from running into blocking problems. Ofcourse that is only relevant if the user does a lot of requests in a short time.

 To prevent the user from running into the blocking it might be helpful to be offensive with the existence of the sleep functionality.